lib/kernel/test/logger_SUITE.erl

```diff
@@ -1009,62 +1010,54 @@ app_config(Config) ->
 %% start, it is not possible to see code coverage in that test.
 kernel_config(Config) ->
     %% Start a node with simple handler only, then simulate kernel
-    %% start by calling internally exported
-    %% internal_init_logger(). This is to test all variants of kernel
-    %% config, including bad config, and see the code coverage.
+    %% start by calling logger:reconfigure(). This is to test all
+    %% variants of kernel config, including bad config, and see
+    %% the code coverage.
     {ok,#{handlers:=[#{id:=simple,filters:=DF}]}=LC,Node} =
         logger_test_lib:setup(Config,[{error_logger,false}]),
 
     %% Same once more, to get coverage
     ok = rpc:call(Node,logger,internal_init_logger,[]),
-    ok = rpc:call(Node,logger,add_handlers,[kernel]),
     LC = rpc:call(Node,logger,get_config,[]),
 
     %% This shall mean the same as above, but using 'logger' parameter
     %% instead of 'error_logger'
     ok = rpc:call(Node,application,unset_env,[kernel,error_logger]),
     ok = rpc:call(Node,application,set_env,
                   [kernel,logger,[{handler,default,undefined}]]),
-    ok = rpc:call(Node,logger,internal_init_logger,[]),
-    ok = rpc:call(Node,logger,add_handlers,[kernel]),
+    ok = rpc:call(Node,logger,reconfigure,[]),
     LC = rpc:call(Node,logger,get_config,[]),
```

Comment:
This testcase fails on this line with a badmatch.

Reply:
Ok, I need to check, as it was passing on my machine last time I checked.

Reply:
Any news on this?

I had a quick look and from what I can tell the problem is that logger:reconfigure does not re-add the simple handler before calling internal_init_logger.

The simple handler is the handler that is used during startup before the entire logging system is started and internal_init_logger assumes that that handler is installed.

If you fix that, then the call to `logger:add_handlers(kernel)` should not have to be made anymore.

Also, don't you need to clear the primary filters before doing internal_init_logger?

Reply:
I now have found some time to fix that. So it should be done soonish.